Support .Net 6

PierreYvesR opened the pull request:
<!-- Please refer to our contributing documentation for any questions on submitting a pull request. -->
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`fake build` or `.\build.cmd`) on local branch was successful 
      `build -t NuGet` works

## Pull request type
<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. --> 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?
Doesn't support .Net 6.

## What is the new behavior?
Supports .Net 6


## Does this introduce a breaking change?
<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->

- [ ] Yes
- [x] No

## Other information
<!-- Any other information that is important to this PR such as screenshots of how the component looks before and after the change. -->

PierreYvesR committed:
Support .Net 6

kerams commented on tests/Fable.SignalR.TestServer/SignalR.fs:
```diff
@@ -1,6 +1,8 @@
 ﻿namespace SignalRApp
 
+#if !NET6_0
```
#if !NET6_0_OR_GREATER
The same problem in 2 other places.

PierreYvesR replied:
fixed. Thanks

PierreYvesR committed:
Use !NET6_0_OR_GREATER instead of !NET6_0 as suggested by @kerams

kerams commented:
I think this should be changed to 2022. The 2019 image on Appveyor does not have .NET 6 and you'd need to install it manually.

https://github.com/Shmew/Fable.SignalR/blob/dab3b002c00cd298d34f6d3c292c735b1fec16ce/appveyor.yml#L3

PierreYvesR committed:
appveyor.yml: use VS2022 - as suggested by @kerams

kerams commented:
Hello, @Shmew, are you around? This would be cool to have.

Trying to add `Fable.SignalR.AspNetCore` (when Giraffe 6 is also used) right now gives me

```
Paket may still find a valid resolution, but this might take a while.
  Conflict detected:
   - fable.signalr.aspnetcore 0.11.5 requested package System.Text.Encodings.Web: >= 5.0 < 6.0
   - System.Text.Json 6.0.0 requested package System.Text.Encodings.Web: >= 6.0
```

I actually wonder if it's really necessary to pin versions of those official .NET packages.

tsujp commented:
@Shmew what's preventing this from being merged? Anything we can help with?

Akash-Mair commented:
@PierreYvesR thanks for getting this across the line! 

I noticed in the `paket.dependencies` the `Net6` group both `Saturn` and `Giraffe` need to be updated to their latest versions, if you want them pinned they need to be updated to `0.16.1` and `6.0.0`

ArtemyB committed:
Build-process and dependencies updates

- Add Fable 4.0.0-theta-018 dotnet tool.
- Add separate Paket dependency groups for Fable demo and tests with Fable.Core 4.0.0-theta-007 and Fable.Elmish 4.
- Update Fable.SignalR.Tests to Elmish 4.
- Update SignalR NPM-package version range (for Femto).
- Update webpack.config.json and package.json (to use Fable 3+ as dotnet tool).
- Add Pnpm for JS package management (could be switched back to Yarn by a couple little build-script changes).
- gitignore and editorconfig updates.

ArtemyB committed:
feat(AspNetCore, Saturn): pass CancellationToken to server-streming functions (`StreamFrom`).

ArtemyB committed:
feat(DotNet-client): pass `CancellationToken` to `StreamFrom` method.

ArtemyB committed:
fix: make `FableHubModule.Stream.Both.Both` constructor public (instead of internal) -- to fix the hub registration error in DI-container; and some code reformatting.

ArtemyB committed:
build: bump version to 0.12.0-alpha.4

ArtemyB committed:
fix(AspNetCore): correct FableHub configuration in `AddSignalR` extension method

Previously all the `OnConnected`, `OnDisconnected`, `Both` hubs have been registered as the `Both` hub specified in the wildcard-case in `Config` match.

ArtemyB committed:
build: bump version to 0.12.0-alpha.5, update release notes

PierreYvesR committed:
Merge pull request #1 from ArtemyB/net6

Merging ArtemyB's changes
